src/main/java/com/faforever/client/achievements/AchievementServiceImpl.java

```diff
@@ -58,9 +52,12 @@ public AchievementServiceImpl(UserService userService, FafService fafService,
 
   @Override
   public CompletableFuture<List<PlayerAchievement>> getPlayerAchievements(Integer playerId) {
-    if (Objects.equals(userService.getUserId(), playerId)) {
+    int currentPlayerId = playerService.getCurrentPlayer().orElseThrow(() -> new IllegalStateException("Player has to be set")).getId();
+    if (Objects.equals(currentPlayerId, playerId)) {
       if (readOnlyPlayerAchievements.isEmpty()) {
-        reloadAchievements();
+        CompletableFuture<List<PlayerAchievement>> achievmentsLoadedFuture = new CompletableFuture<>();
+        reloadAchievements(achievmentsLoadedFuture);
+        return achievmentsLoadedFuture;
```

Comment:
`return reloadAchievements();` and have the method create the future.
And in `postConstruct()` you can just ignore the return value.